### PR TITLE
fix: use headers as part of generating new queries

### PR DESCRIPTION
**Pull request by jrea**

### CODE REVIEW

**Review by ElizabethSchafle — APPROVED** [review]
💯

File: packages/redux-query-react/src/components/connect-request.js [hunk 1]
```diff
@@ -54,6 +54,25 @@ const diffQueryConfigs = (
   return { cancelKeys, requestQueryConfigs };
 };
 
+export const headersChanged = (
+  queryConfigs: Array<QueryConfig>,
+  previousQueryConfigs: Array<QueryConfig>,
+) => {
+ return queryConfigs.map(function (config, i) {
+    if (previousQueryConfigs[i] && queryConfigs[i] && previousQueryConfigs[i].options && queryConfigs[i].options) {
+      var prevHeaders = previousQueryConfigs[i].options.headers;
```
Comment (dylanharrington) [inline packages/redux-query-react/src/components/connect-request.js]: let/const should work here, right?

File: packages/redux-query-react/src/components/connect-request.js [hunk 2]
```diff
@@ -54,6 +54,25 @@ const diffQueryConfigs = (
...
+      var headers = queryConfigs[i].options.headers;
+
+      if (prevHeaders != null && headers != null) {
+        var prevHeaderValues = Object.values(prevHeaders);
+        return Object.values(headers).some(function (value, i) {
+          return prevHeaderValues[i] !== value;
```
Comment (dylanharrington) [inline packages/redux-query-react/src/components/connect-request.js]: can it ever be `null === undefined`? but i think this is right

**Review by dylanharrington — APPROVED** [review]

File: packages/redux-query-react/src/components/connect-request.js (hunk 1 above)
Comment (jrea) [reply packages/redux-query-react/src/components/connect-request.js]: yes, copy pasta from the already built file

File: packages/redux-query-react/src/components/connect-request.js (hunk 2 above)
Comment (jrea) [reply packages/redux-query-react/src/components/connect-request.js]: I think this might be possible, updated and added a test case

**Review by dylanharrington — APPROVED** [review]
nice

**Pull request merged**